modules/fedimint-ln-server/src/lib.rs

```diff
@@ -859,6 +898,25 @@ impl Lightning {
         dbtx.insert_entry(&LightningGatewayKey(gateway.node_pub_key), &gateway)
             .await;
     }
+
+    async fn process_clock_time_proposals(&self, mut proposals: Vec<SystemTime>) -> SystemTime {
+        assert!(!proposals.is_empty());
+
+        proposals.sort();
+        let median_proposal = match proposals.len() % 2 {
+            0 => {
+                let earlier_time = proposals[proposals.len() / 2 - 1];
+                let later_time = proposals[proposals.len() / 2];
+                let time_diff = later_time.duration_since(earlier_time);
+                earlier_time
+                    + time_diff
+                        .expect("Error calculating median time")
+                        .div_f32(2.0)
```

Comment:
I'd rather avoid floating point operations since they tend to slightly differ on some architectures (mostly due to CPU bugs afaik).